src/checkfiles/checkfiles.py

```diff
@@ -0,0 +1,197 @@
+    'KEY', '2022/10/31/8b19341b-b1b2-4e10-ad7f-aa910ccd4d2c/ENCFF206HGF.bam')
+MD5SUM = os.getenv('MD5SUM', '3e814f4af7a4c13460584b26fbe32dc4')
+FILE_FORMAT = os.getenv('FILE_FORMAT', 'bam')
+UUID = os.getenv('UUID', '462bd56-9278-48aa-bc55-9eff587ba2c7')
+FILE_SIZE = os.getenv('FILE_SIZE', 1371)
+NUMBER_OF_READS = os.getenv('NUMBER_OF_READS', 25)
+READ_LENGTH = os.getenv('READ_LENGTH', 58)
+DATA_DIR = '/s3/'
+CHUNK_SIZE = 128*6400
+
+
+ZIP_FILE_FORMAT = [
+    'bam',
+    'fastq',
+    'txt',
+    'tsv',
+]
+
+EXCLUDE_FORMAT = [
+    'bai',
+]
+
+CONTENT_MD5SUM_URL = 'https://www.encodeproject.org/search/?type=File&format=json&content_md5sum='
+
+logging.basicConfig(
+    format='%(asctime)s | %(levelname)s: %(message)s', level=logging.INFO)
+
+
+def main():
+    try:
+        response = file_validation(BUCKET_NAME, KEY, UUID,
+                                   MD5SUM, FILE_FORMAT, FILE_SIZE, NUMBER_OF_READS, READ_LENGTH)
+        logging.info(json.dumps(response))
+    except Exception as err:
+        message = f'exception occurred when checking file uuid #{UUID}: {str(err)}'
+
+        logging.info(json.dumps({'exception': message}))
+        sys.exit(1)  # Retry Job Task by exiting the process
+
+
+def file_validation(bucket_name, key, uuid, md5sum, file_format, file_size, number_of_reads, read_length):
+    logging.info(f'Checking file uuid {uuid}...')
+    response = boto3.client('s3').get_object(Bucket=bucket_name, Key=key)
+    file_path = get_local_file_path(key)
+
+    errors = {}
+    results = {}
+
+    is_gzipped = is_file_gzipped(file_path)
+    logging.info(f'is file gziped: {is_gzipped}')
+    check_valid_gzipped_file_format(errors, is_gzipped, file_format)
+    results['file_size'] = response.get('ContentLength')
+    check_file_size(errors, file_size, results['file_size'])
+    check_md5sum(errors, md5sum, response.get('ETag').strip('"'), file_path)
+
+    if is_gzipped:
+        check_content_md5sum(errors, file_path)
+
+    if file_format == 'bam':
+        bam_pysam_check(errors, file_path, number_of_reads)
+        if 'bam_error' not in errors:
+            bam_generate_bai_file(file_path)
+    elif file_format == 'fastq':
+        fastq_check(errors, file_path, number_of_reads, read_length)
+
+    logging.info(f'Completed file validation for file uuid {uuid}.')
+
+    if errors:
+        return {
+            'uuid': uuid,
+            'validation_result': 'failed',
+            'errors': errors
+        }
+    else:
+        return {
+            'uuid': uuid,
+            'validation_result': 'pass'
+        }
+
+
+def get_local_file_path(blob_name):
+    file_path = DATA_DIR + blob_name
+    return file_path
+
+
+def is_file_gzipped(file_path):
+    try:
+        gzip.GzipFile(filename=file_path).read(1)
+        return True
+    except gzip.BadGzipFile:
+        return False
+
+
+def check_valid_gzipped_file_format(errors, is_gzipped, file_format):
+    if file_format in ZIP_FILE_FORMAT and not is_gzipped:
```

Comment:
In general I think this kind of "silent" dependency on a constant is suspect. Would be better to pass as parameter I think.

Reply:
fixed